python/ray/dag/compiled_dag_node.py

```diff
@@ -1238,6 +1235,8 @@ def _get_or_compile(
                         assert len(upstream_task.output_channels) == 1
                         arg_channel = upstream_task.output_channels[0]
                         assert arg_channel is not None
+                if is_input_consumer:
+                    num_input_consumers += 1
```

Comment:
I think we will need to compute how many tasks that consume the input, so there needs some dedup if the tasks are the same.

Say for task1(inp[0], inp[1]), it should only be count as 1

Reply:
> Say for task1(inp[0], inp[1]), it should only be counted as 1

Is there any difference from the current behavior? The current logic is:

```python
for task in tasks:
    is_input_consumer = False
    for arg in task.args:
      # update `is_input_consumer`
    if is_input_consumer:
      num_input_consumers += 1
```

`num_input_consumers` at most increments by 1 for a task.